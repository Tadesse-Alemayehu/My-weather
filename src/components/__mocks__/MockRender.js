import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../Redux/configureStore';

export default (component) => (
  <Provider store={store}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>
);
