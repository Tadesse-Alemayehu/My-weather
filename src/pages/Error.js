import React from 'react'
import { useSearchParams, useNavigate } from 'react-router-dom';
export default function Error() {
  const [getParam, setParam] = useSearchParams();
  const errorMessage = getParam.get('message');
  const navigate = useNavigate();
  return (
    <div id="error" style={{ "text-align": "center" }}>
      <h1>Error</h1>
      <h3>{errorMessage}</h3>
      <button className="home" onClick={() => navigate('/')}>Home</button>
    </div>
  )
}
