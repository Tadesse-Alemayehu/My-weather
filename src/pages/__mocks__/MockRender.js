import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../Redux/configureStore';

const MockRender = (component) => (
  <Provider store={store}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>
);

export default MockRender;
