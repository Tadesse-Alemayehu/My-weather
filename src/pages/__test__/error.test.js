import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import MockRender from '../__mocks__/MockRender';
import Error from '../Error';

describe('error page test', () => {
  test('Should display the error message', () => {
    render(
      MockRender(<Error />),
    );
    const error = screen.getByText(/error/i);
    expect(error).toBeInTheDocument();
  });
});

describe('Snapshot the check for the Error page', () => {
  test('should match the snapshot', () => {
    const errorPageSnap = renderer
      .create(MockRender(<Error />)).toJSON();
    expect(errorPageSnap).toMatchSnapshot();
  });
});
