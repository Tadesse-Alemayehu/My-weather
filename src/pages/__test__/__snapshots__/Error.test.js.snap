// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot the check for the Error page should match the snapshot 1`] = `
<div
  id="error"
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <h1>
    Error
  </h1>
  <h3 />
  <button
    className="home"
    onClick={[Function]}
  >
    Home
  </button>
</div>
`;
